### PR TITLE
Update all the things

### LINKED ISSUE #4: upgrade to use broccoli-filter v1.3.0
I have made an updated version using broccoli-filter v1.3.0 and using broccoli >= V1.x for the tests.
Can i make a pull request?

### PR DESCRIPTION
**Pull request by simonihmig**
Updates all dependencies, most importantly a recent version of `broccoli-filter` (closes #4), introduces modern JS syntax, and bumps the required node version to >= 10. 

This is breaking change, but I figured after 7y since the last commit, this *might* be acceptable!? 😜

### CODE REVIEW

**Comment by simonihmig** [comment]
See https://github.com/emberjs/ember-classic-decorator/issues/63

**Comment by rwjblue** [comment]
Haha, its been soooo long!!!!

**Comment by rwjblue** [comment]
We could also swap to GH Actions

File: index.js [hunk 1]
```diff
@@ -1,36 +1,32 @@
...
-  if (!(this instanceof Derequire)) return new Derequire(inputTree, options);
-
-  Filter.call(this, inputTree, options); // this._super()
-
-  options = options || {};
-
-  this.inputTree   = inputTree;
-  this.files       = options.files || [];
-  this.description = options.description;
+const Filter = require('broccoli-filter');
```
Comment (rwjblue) [inline index.js]: Any reason not to swap to broccoli-persistent-filter?

File: index.js [hunk 2]
```diff
@@ -1,36 +1,32 @@
...
+  processString(str) {
+    return derequire(str, this.patterns);
   }
-};
-Derequire.prototype = Object.create(Filter.prototype);
-Derequire.prototype.constructor = Derequire;
+}
 
 Derequire.prototype.extensions = ['js'];
 Derequire.prototype.targetExtension = 'js';
```
Comment (rwjblue) [inline index.js]: these can be class fields that you set in the constructor

**Comment by simonihmig** [comment]
> We could also swap to GH Actions

Sure, I can do that. But didn't want to put too much into a single PR.

File: index.js (hunk 1 above)
Comment (simonihmig) [reply index.js]: Hm, no... will update.

File: index.js (hunk 2 above)
Comment (simonihmig) [reply index.js]: Are they already available in node10?

Comment (rwjblue) [reply index.js]: Sorry, I was meaning in the constructor (you are correct class fields are not available in node 10).

```js
class Thing extends Filter {
  constructor() {
    super(...stuff);
    this.extensions = ['js']';
    this.targetExtensions = ['js'];
  }
}
```

Comment (simonihmig) [reply index.js]: Ah ok. Yeah, I remember I was not sure if they *must* live on the prototype as before or also on the instance, so went for the most compatible change. But if everything works this way also, then I will update this.

**Comment by simonihmig** [comment]
@rwjblue addressed your feedback! (thanks for the super quick response, btw!)

**Review by rwjblue — APPROVED** [review]

**Comment by simonihmig** [comment]
BTW, will also need to update the `ember-derequire` counterpart after this is done. So would appreciate a new release! :)

**Pull request merged**